### DISCUSSION TITLE
Treat ref struct fields as fixed statements

### DISCUSSION THREAD

**Discussion by Avid29** [dotnet/csharplang#7268] [Language Ideas]
Currently if you attempt to make a pointer to a `ref struct` field it will result in an error `CS0212: You can only take the address of an unfixed expression inside of a fixed statement initializer`. However, because of the properties of a `ref struct` its members are not in managed memory and therefore are effectively fixed.

Example code:
```cs
public ref struct ExampleStruct
{
    private int _bar;
    private int _foo;
    
    public unsafe void FooBar(bool foo)
    {
        int* foobarPtr = foo switch
        {
            true => &_bar,   // Error CS0212
            false => &_foo,  // Error CS0212
        };
    }
}
```

Again, this code will give an error ``CS0212`` because `_bar` and `_foo` are not explicitly pinned, however they are guaranteed not to move.

I propose that `_bar` and `_foo`, or any `ref struct` field, would be treated as fixed in order to allow pointers.

**Comment by 333fred** [comment]
Duplicate of https://github.com/dotnet/csharplang/issues/1792.
